File: app/src/main/java/com/ebr163/inspector/sample/MainActivity.java
package com.ebr163.inspector.sample;

import android.app.Activity;
import android.os.Bundle;
import android.widget.EditText;

import com.ebr163.inspector.Inspection;
import com.ebr163.inspector.Inspector;
import com.ebr163.inspector.rule.TextNotEmptyRule;
import com.ebr163.inspector.view.InspectViewBuilder;
import com.ebr163.inspector.view.InspectionView;

public class MainActivity extends Activity {

    private Inspector inspector;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        inspector = new Inspector();

        Inspection<String> inspectionView = new InspectViewBuilder< EditText, String>(new EditText(this))
                .addRule(new TextNotEmptyRule("text is empty"))
                .addValueListener(new InspectionView.OnValueListener<EditText, String>() {
                    @Override
                    public String getValue(EditText view) {
                        return view.getText().toString();
                    }
                })
                .addErrorListener(new InspectionView.OnErrorListener<EditText>() {
                    @Override
                    public void setErrorEnabled(EditText view, String error, boolean enabled) {
                        //do something
                    }
                })
                .build();
    }

    @Override
    protected void onDestroy() {
        inspector.clear();
        super.onDestroy();
    }
}

File: app/src/main/java/com/ebr163/inspector/sample/MainActivity.kt
package com.ebr163.inspector.sample

import android.os.Bundle
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.ebr163.inspector.Inspector
import com.ebr163.inspector.rule.TextNotEmptyRule
import com.ebr163.inspector.variable.InspectionVariable
import com.ebr163.inspector.variable.inspectionVariable
import com.ebr163.inspector.view.InspectViewBuilder

class MainActivity : AppCompatActivity() {

    private var inspector: Inspector? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        inspector = Inspector(this.lifecycle)

        InspectViewBuilder<EditText, String>(EditText(this))
                .addRule(TextNotEmptyRule("text is empty"))
                .addValueListener {
                    it.text.toString()
                }
                .addErrorListener { _, _, _ ->
                    //do something
                }
                .build()

        val inspection: InspectionVariable<String> = inspectionVariable {
            value = "Test"
            rule(TextNotEmptyRule(""))
            onErrorListener { enabled, errorMessage ->
                // do something
            }
        }
        inspector?.addInspection(inspection)
    }

    override fun onDestroy() {
        inspector?.clear()
        super.onDestroy()
    }
}
